builtin/providers/aws/resource_aws_db_parameter_group_test.go

```diff
@@ -106,6 +108,60 @@ func TestAccAWSDBParameterGroupOnly(t *testing.T) {
 	})
 }
 
+func TestResourceAWSDBParameterGroupName_uppercaseCharacter(t *testing.T) {
+	var dbParamGroupName = "tEsting123"
+	_, errors := validateDbParamGroupName(dbParamGroupName, "aws_db_parameter_group_name")
+
+	if len(errors) != 1 {
+		t.Fatalf("Expected the DB Parameter Group Name to trigger a validation error")
+	}
+}
```

Comment:
Lots of tests! Since they're all testing the same method with different inputs, can we re-write this to be a table test?

```go
func TestResourceAWSDBParameterGroupName_validation(t *testing.T) {
	cases := []struct {
		Value    string
		ErrCount int
	}{
		{
			Value:    "tEsting123",
			ErrCount: 1,
		},
		{
			Value:    "testing123!",
			ErrCount: 1,
		},
               # [.. so on ..]
	}

	for _, tc := range cases {
		_, errors := validateDbParamGroupName(tc.Value, "SampleKey")
		if len(errors) != ErrCount {
			t.Fatalf("Expected the DB Parameter Group Name to trigger a validation error")
		}
	}
}
```